python/tempo/resample.py

```diff
@@ -24,24 +27,24 @@ def __appendAggKey(tsdf, freq = None):
     """
     :param tsdf: TSDF object as input
     :param freq: frequency at which to upsample
-    :return: return a TSDF with a new aggregate key (called agg_key)
+    :return: triple - 1) return a TSDF with a new aggregate key (called agg_key) 2) return the period for use in interpolation, 3) return the time increment (also necessary for interpolation)
     """
     df = tsdf.df
     parsed_freq = checkAllowableFreq(tsdf, freq)
     agg_window = f.window(f.col(tsdf.ts_col), "{} {}".format(parsed_freq[0], freq_dict[parsed_freq[1]]))
 
     df = df.withColumn("agg_key", agg_window)
-    return tempo.TSDF(df, tsdf.ts_col, partition_cols = tsdf.partitionCols)
+    return (tempo.TSDF(df, tsdf.ts_col, partition_cols = tsdf.partitionCols), parsed_freq[0], freq_dict[parsed_freq[1]])
 
-def aggregate(tsdf, freq, func, metricCols = None, prefix = None):
+def aggregate(tsdf, freq, func, metricCols = None, prefix = None, fill = None):
     """
     aggregate a data frame by a coarser timestamp than the initial TSDF ts_col
     :param tsdf: input TSDF object
     :param func: aggregate function
     :param metricCols: columns used for aggregates
```

Comment:
Can we document additional params:
- prefix
- fill

Reply:
Thanks